### ISSUE TITLE
Uncloaking based on unit HP is hard coded

### ISSUE THREAD

**Issue by forcecore** [OpenRA/OpenRA#12741]
Hi, I'm trying to make ants keep burrowed to the ground even when the ant is in critical damage state. It seems the related code is hardcoded.

https://github.com/OpenRA/OpenRA/blob/285e9a803067ddf8bbfee3b89f487029d6118833/OpenRA.Mods.Common/Traits/Cloak.cs#L106

```
void INotifyDamageStateChanged.DamageStateChanged(Actor self, AttackInfo e)
{
    damageDisabled = e.DamageState >= DamageState.Critical;
    if (damageDisabled || Info.UncloakOn.HasFlag(UncloakType.Damage))
        Uncloak();
}
```

No SC1 Dark Templars for OpenRA :(

**Comment by forcecore** [comment]
UncloakType.Damage seems ambiguous to me. If it is supposed to reveal a clocked unit when the unit has been hit by something, then a better name would be OnHit or OnDamage.

On the other hand, if it was supposed to mean damage state, then || operator should become &&, and the could be renamed to CriticalDamageState

**Comment by pchote** [comment]
The `damageDisabled` bit can be removed completely, mods that want to disable based on damage state can do this using conditions.

**Comment by forcecore** [comment]
Sounds like an easy fix. I'll branch from bleed and submit a PR when done. Both Cloak.cs and YAML files.

**Comment by forcecore** [comment]
PR #12759 accepted, closing.

**Comment by pchote** [comment]
Note that you can make GitHub close issues automatically by putting e.g. "Fixes #12741" in a PR description.
